tests/parsec-cli-tests.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env sh
+
+# Copyright 2021 Contributors to the Parsec project.
+# SPDX-License-Identifier: Apache-2.0
+
+# Run simple end-two-end Parsec tests using parsec-tool and openssl
+
+ping_parsec() {
+    echo "Checking Parsec service... "
+    $PARSEC_TOOL ping
+}
+
+run_cmd() {
+    "$@"
+    EXIT_CODE=$(($EXIT_CODE+$?))
+}
+
+debug() {
+    if [ -n "$PARSEC_TOOL_DEBUG" ]; then
+        "$@"
+    fi
+}
+
+test_crypto_provider() {
+# $1 - provider ID
+
+    PARSEC_TOOL_CMD="$PARSEC_TOOL -p $1"
+
+    echo
+    echo "- Test random number generation"
+    if run_cmd $PARSEC_TOOL_CMD list-opcodes 2>/dev/null | grep -q "PsaGenerateRandom"; then
+        run_cmd $PARSEC_TOOL_CMD generate-random --nbytes 10
+    else
+        echo "This provider doesn't support random number generation"
+    fi
+
+    test_rsa
+    test_ecc
+}
+
+test_rsa() {
+    KEY="anta-key-rsa"
+    KEY_PARAM="--key-name $KEY"
+    TEST_STR="$(date) Parsec decryption test"
+
+    echo
+    echo "- Creating an RSA key and exporting its public part"
+    run_cmd $PARSEC_TOOL_CMD create-rsa-key $KEY_PARAM
+    run_cmd $PARSEC_TOOL_CMD list-keys
```

Comment:
Should this include a `| grep $KEY`?

Reply:
Initially I thought it's not required because we're checking exit codes for create, list and export key commands. But, we could miss a case when a key was successfully created and exported and at the same time for whatever reason was missing in the list although the exit code was ok. I'll add a check